src/main/java/org/variantsync/diffdetective/datasets/predefined/Marlin.java

```diff
@@ -13,14 +13,16 @@
  * @author Kevin Jedelhauser, Paul Maximilian Bittner
  */
 public class Marlin {
-    public static final CPPAnnotationParser ANNOTATION_PARSER =
-        new CPPAnnotationParser(
-            PropositionalFormulaParser.Default,
-            new MarlinCPPDiffLineFormulaExtractor()
-        );
+    public static final PreprocessorAnnotationParser ANNOTATION_PARSER =
+            new PreprocessorAnnotationParser(
+                    PreprocessorAnnotationParser.CPP_PATTERN,
+                    PropositionalFormulaParser.Default,
+                    new MarlinCPPDiffLineFormulaExtractor()
+            );
```

Comment:
Instantiating a parser like this looks like it leaks implementation details. In particular, to a new user it might not be clear which patterns to pass here? Would it make sense to introduce some smart constructors (i.e., static methods such as `CreateCPPAnnotationParser` and `CreateJPPAnnotationParser`, which fix / curry the `XPP_PATTERN`)? Not too important though.